src/actions/coordinator.ts

```diff
@@ -84,20 +78,34 @@ export function CoordinatorActions<ActionsBase extends Constructor<TinlakeParams
       return this.pending(coordinator.executeEpoch())
     }
 
+    getCurrentEpochId = async () => {
+      const coordinator = this.contract('COORDINATOR')
+      return (await coordinator.currentEpoch()).toBN().toNumber()
+    }
+
+    getCurrentEpochMinimumTimeEnd = async () => {
+      const coordinator = this.contract('COORDINATOR')
+
+      const lastEpochClosed = (await coordinator.lastEpochClosed()).toBN().toNumber()
+      const minimumEpochTime = (await coordinator.minimumEpochTime()).toBN().toNumber()
+
+      return lastEpochClosed + minimumEpochTime
+    }
+
     getCurrentEpochState = async () => {
       const coordinator = this.contract('COORDINATOR')
 
-      const lastEpochClosed =  (await coordinator.lastEpochClosed).toBN().toNumber()
-      const minimumEpochTime = (await coordinator.minimumEpochTime).toBN().toNumber()
-      if (((new Date()).getTime() - lastEpochClosed) >= minimumEpochTime) {
+      const lastEpochClosed = (await coordinator.lastEpochClosed()).toBN().toNumber()
+      const minimumEpochTime = (await coordinator.minimumEpochTime()).toBN().toNumber()
+      if (new Date().getTime() - lastEpochClosed >= minimumEpochTime) {
```

Comment:
This would be unreliable if the system time is not correct. Can we instead compare the time with the timestamp of the latest ethereum block?

Reply:
Good point! I've addressed this in #203.